formats/hocon/src/test/kotlin/kotlinx/serialization/hocon/HoconValuesTest.kt

```diff
@@ -103,4 +138,29 @@ class HoconValuesTest {
             assertEquals(listOf(null, WithNullable(10, "bar")), i3)
         }
     }
+
+    @Test
+    fun `deserialize list of integer string values`() {
+        val configString = """i1 = [ "1","3" ]"""
+        val obj = deserializeConfig(configString, WithList.serializer())
+        assertEquals(listOf(1, 3), obj.i1)
+    }
+
+    @Test
+    fun `deserialize map with integers`() {
+        val configString = """m = { 2: 1, 4: 3 }"""
+        val obj = deserializeConfig(configString, WithMap.serializer())
+        with(obj) {
+            assertEquals(mapOf(2 to 1, 4 to 3), m)
+        }
+    }
+
+    @Test
+    fun `deserialize map with integers as strings`() {
+        val configString = """m = { "2": "1", "4":"3" }"""
+        val obj = deserializeConfig(configString, WithMap.serializer())
+        with(obj) {
```

Comment:
and here

Reply:
:+1: sorry for not having checked this thoroughly before.